packages/common/src/ContractUtils.js

```diff
@@ -97,8 +99,19 @@ const runTransaction = async ({ transaction, config }) => {
     throw error;
   }
 };
-
-module.exports = {
-  revertWrapper,
-  runTransaction
+/**
+ * Blocking code until a specific block number is mined. Will re-fetch the current block number every 500ms. Useful when
+ * using methods called on contracts directly after state changes. Max blocking time should be ~ 15 seconds.
+ * @param {Object} web3 Provider from Truffle/node to connect to Ethereum network.
+ * @param {number} blockerBlockNumber block execution until this block number is mined.
+ */
+const blockUntilBlockMined = async (web3, blockerBlockNumber) => {
+  if (argv._.indexOf("test") !== -1) return;
+  for (;;) {
+    const currentBlockNumber = await web3.eth.getBlockNumber();
+    if (currentBlockNumber >= blockerBlockNumber) break;
+    await new Promise(r => setTimeout(r, 500));
+  }
```

Comment:
Could this be done using: https://web3js.readthedocs.io/en/v1.2.11/web3-eth-subscribe.html#subscribe-newblockheaders? Or does that not work with http connections?

Reply:
you could totally do it like that but I was not sure how to structure the method to _block_ until it had mined the associated block number. I tried using the `.on("data", function(blockHeader){...` syntax to toggle a flag and then the method would return but I could not get it to behave as expected. 

Perhaps you can think of a pattern to do this? I agree a subscription would be better than polling like I'm doing here.

Reply:
I think the for loop is a bit easier to reason about, because you `await` this entire method in `_executeLiquidationWithDsProxy`

Reply:
yup, exactly. only trade off it is could be bit slower (up to half a second) and potentially has more web3 calls.